sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_test.cpp

```diff
@@ -572,7 +572,25 @@ TEST_F(KeyVaultCertificateClientTest, BackupRestoreCertificate)
     auto result = response.PollUntilDone(m_defaultWait);
     EXPECT_EQ(result.Value.Name(), certificateName);
     client.PurgeDeletedCertificate(certificateName);
-    std::this_thread::sleep_for(m_defaultWait);
+    TestSleep(m_defaultWait);
+  }
+  {
+    // before we restore we need to ensure the certificate is purged.
+    // since we have no visibility into the purge status we will just wait and check for a few
+    // seconds
+    while (true)
```

Comment:
WHat happens if this never throws?

Would it make sense to (say) try 15 times and fail the test after 15 iterations rather than continue to try (which would hang the test job)>?

Reply:
then the test would fail falsely. i can add a certain number of retries but it would still not solve the issue unless we wait. purge can take any amount of time,. Since the pipelines times out eventually it will exit.